### DISCUSSION TITLE
withPivot('*') ?

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#33487] [General]
I'm looking for something that allow me to pick all the extra pivot columns like `withPivot('*')` or `withAllPivot()`, I hate doing this : 

```php
return $this->belongsToMany(Event::class, 'events_awards')
            ->withPivot(['column1', 'column2', 'column3', 'column4', column5, column6]);
```

If not, Can anybody make it happen? I'm not good at framework

**Comment by ryangjchandler** [comment]
Something like this isn't possible since the column names need to be aliases, therefore Laravel needs to know the column names. See the discussion at #23298.
